src/ore/src/assert.rs

```diff
@@ -107,6 +107,27 @@ macro_rules! soft_assert_or_log {
     }}
 }
 
+/// Asserts that two expressions are equal to each other if soft assertions
+/// are enabled, or logs an error if soft assertions are disabled and the
+/// two expressions are not equal.
+#[macro_export]
+macro_rules! soft_assert_eq_or_log {
+    ($left:expr, $right:expr) => {{
+        if $crate::assert::SOFT_ASSERTIONS.load(::std::sync::atomic::Ordering::Relaxed) {
+            assert_eq!($left, $right);
+        } else if $left != $right{
+            ::tracing::error!("assertion {:?} == {:?} failed", $left, $right)
+        }
+    }};
+    ($left:expr, $right:expr, $($arg:tt)+) => {{
+        if $crate::assert::SOFT_ASSERTIONS.load(::std::sync::atomic::Ordering::Relaxed) {
+            assert_eq!($left, $right, $($arg)+);
+        } else if $left != $right{
+            ::tracing::error!("assertion {:?} == {:?} failed: {}", $left, $right, format!($($arg)+))
+        }
+    }};
+}
+
```

Comment:
@ParkMyCar Can you take a look at this? `soft_assert_or_log` didn't really match the semantics that I was looking for.

Reply:
Looks good to me, thanks for adding this!